apps/portal-api/src/modules/services/definition/service-definition.resolver.test.ts

```diff
@@ -0,0 +1,113 @@
+          name: 'Capability 1',
+          description: null,
+          service_definition_id: null,
+        },
+        {
+          id: toGlobalId('Service_Capability', '2') as ServiceCapabilityId,
+          name: 'Capability 2',
+          description: null,
+          service_definition_id: null,
+        },
+      ];
+
+      const mockLoadServiceCapabilitiesBy = vi
+        .spyOn(ServiceCapabilityDomain, 'loadServiceCapabilitiesBy')
+        .mockResolvedValueOnce(mockServiceCapabilities);
+      const resolver =
+        serviceDefinitionResolver.ServiceDefinition?.service_capability;
+      if (!resolver) {
+        throw new Error('service_capability resolver is not defined');
+      }
+
+      const response = await resolver(
+        { id: mockServiceDefinitionId } as unknown as ServiceDefinition,
+        {},
+        contextAdminUser,
+        {} as GraphQLResolveInfo
+      );
+
+      expect(mockLoadServiceCapabilitiesBy).toHaveBeenCalledWith(
+        contextAdminUser,
+        {
+          service_definition_id: mockServiceDefinitionId,
+        }
+      );
+      expect(response).toEqual(mockServiceCapabilities);
+    });
+
+    it('should handle empty service capabilities', async () => {
+      const mockServiceDefinitionId =
+        'test-service-def-id' as ServiceDefinitionId;
+
+      const mockLoadServiceCapabilitiesBy = vi
+        .spyOn(ServiceCapabilityDomain, 'loadServiceCapabilitiesBy')
+        .mockResolvedValueOnce([]);
+
+      const resolver =
+        serviceDefinitionResolver.ServiceDefinition?.service_capability;
+      if (!resolver) {
+        throw new Error('service_capability resolver is not defined');
+      }
+
+      const response = await resolver(
+        { id: mockServiceDefinitionId } as unknown as ServiceDefinition,
+        {},
+        contextAdminUser,
+        {} as GraphQLResolveInfo
+      );
+
+      expect(response).toEqual([]);
+      expect(mockLoadServiceCapabilitiesBy).toHaveBeenCalledOnce();
+    });
+
+    it('should handle errors from loadServiceCapabilitiesBy', async () => {
+      const mockServiceDefinitionId =
+        'test-service-def-id' as ServiceDefinitionId;
+      const mockError = new Error('Failed to load capabilities');
+
+      vi.spyOn(
+        ServiceCapabilityDomain,
+        'loadServiceCapabilitiesBy'
+      ).mockRejectedValue(mockError);
+
+      const resolver =
+        serviceDefinitionResolver.ServiceDefinition?.service_capability;
+      if (!resolver) {
+        throw new Error('service_capability resolver is not defined');
+      }
+
+      await expect(
+        resolver(
+          { id: mockServiceDefinitionId } as unknown as ServiceDefinition,
+          {},
+          contextAdminUser,
+          {} as GraphQLResolveInfo
+        )
+      ).rejects.toThrow('Failed to load capabilities');
+    });
+  });
+});
+
+afterEach(async () => {
+  vi.clearAllTimers();
+  vi.restoreAllMocks();
+  vi.clearAllMocks();
+});
```

Comment:
This block should be in the first `describe` block. I always recommend to put it at the beginning of the block because it's easier to know it exists before reading tests.